x/cronos/rpc/api.go

```diff
@@ -68,51 +74,61 @@ func NewCronosAPI(
 		panic(err)
 	}
 	return &CronosAPI{
-		ctx:          context.Background(),
-		clientCtx:    clientCtx,
-		queryClient:  rpctypes.NewQueryClient(clientCtx),
-		chainIDEpoch: eip155ChainID,
-		logger:       logger.With("client", "json-rpc"),
-		backend:      backend,
+		ctx:               context.Background(),
+		clientCtx:         clientCtx,
+		queryClient:       rpctypes.NewQueryClient(clientCtx),
+		chainIDEpoch:      eip155ChainID,
+		logger:            logger.With("client", "json-rpc"),
+		backend:           backend,
+		cronosQueryClient: types.NewQueryClient(clientCtx),
 	}
 }
 
-// GetTransactionReceiptsByBlock returns all the transaction receipts included in the block.
-func (api *CronosAPI) GetTransactionReceiptsByBlock(blockNrOrHash rpctypes.BlockNumberOrHash) ([]map[string]interface{}, error) {
-	var receipts []map[string]interface{}
-
-	api.logger.Debug("cronos_getTransactionReceiptsByBlock", "blockNrOrHash", blockNrOrHash)
-
-	blockNum, err := api.getBlockNumber(blockNrOrHash)
+func (api *CronosAPI) GetBlockDetail(blockNrOrHash rpctypes.BlockNumberOrHash) (
+	resBlock *coretypes.ResultBlock,
+	blockNumber int64,
+	blockHash string,
+	blockRes *coretypes.ResultBlockResults,
+	baseFee *big.Int,
+	err error,
```

Comment:
why we are returning  `blockNumber ` and `blockHash `, the `resBlock` already contains these data.
Or should we create a new struct to return the data the client requires instead return everything?

Reply:
If this API is only called by `GetTransactionReceiptsByBlock `, should we not expose this API to the public?

Reply:
done, changed to private.